### ISSUE TITLE
Icon matchTextDirection not work with rtl while using impeller

### ISSUE THREAD

**Issue by Elblassy** [flutter/flutter#153036]
### Steps to reproduce

1- put any icon 
2- run app with impeller 
3- change to arabic language or any language with rtl 

### Expected results

Icon match the direction of text 

### Actual results

When i update flutter to flutter 3.24 i found issue with impeller that icons with rtl not mirrored 
<img width="344" alt="Screenshot 2024-08-07 at 8 08 20 PM" src="https://github.com/user-attachments/assets/a3f41b48-9c57-48cc-9997-670cc6490354">

### Code sample

<details open><summary>Code sample</summary>

```dart
IconButton(
                  onPressed: () {
                    context.pop();
                  },
                  icon: const Icon(Icons.arrow_back_ios_new),
                ),
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.0, on macOS 14.5 23F79 darwin-arm64, locale
    en-EG)
    • Flutter version 3.24.0 on channel stable at
      /Users/elblasy/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 80c2e84975 (8 days ago), 2024-07-30 23:06:49 +0700
    • Engine revision b8800d88be
    • Dart version 3.5.0
    • DevTools version 2.37.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/elblasy/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/elblasy/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.11+0-17.0.11b1207.24-11852314)

[✓] IntelliJ IDEA Community Edition (version 2023.3.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.87.2)
    • VS Code at /Users/elblasy/Downloads/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • iPhone 15 Pro Max (mobile)      • 620ED9A3-77A7-4F89-8ECF-EAABBCD60239 •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-2 (simulator)
    • macOS (desktop)                 • macos                                •
      darwin-arm64   • macOS 14.5 23F79 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                •
      darwin         • macOS 14.5 23F79 darwin-arm64
    • Chrome (web)                    • chrome                               •
      web-javascript • Google Chrome 127.0.6533.89
    ! Error: Browsing on the local area network for Aya❤️. Ensure the device is
      unlocked and attached with a cable or associated with the same local area
      network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code
      -27)
    ! Error: Browsing on the local area network for Yahia’s iPhone. Ensure the
      device is unlocked and attached with a cable or associated with the same
      local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code
      -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!


```

</details>

**Comment by wasilni2022** [comment]
same here!!

**Comment by joelrose2** [comment]
I am seeing the same thing since upgrading to Flutter 3.24.0. There are actually two problems: the one mentioned in this issue (directional icons are not being flipped in iOS), and also the other issue mentioned in https://github.com/flutter/flutter/issues/153133 (the position of the icon is incorrectly shifted to the right). 

In the meantime, the following workaround works for iOS and doesn't harm Android: Set the icon as **textDirection: TextDirection.ltr** AND apply a 180-degree rotation when UI is in an RTL language: **Transform.rotate(angle: isRTL ? 3.1416 : 0, child: Icon(...**

**Comment by saleh-aleid** [comment]
any updates?

**Comment by yazanturk** [comment]
any update ?

**Comment by moshe5745** [comment]
Title need to be updated
The issue is iOS only

**Comment by mNouh94** [comment]
This issue has been fixed during update [#3.24.1](https://github.com/flutter/flutter/blob/stable/CHANGELOG.md#3241)

**Comment by jonahwilliams** [comment]
C
